pasqal_cloud/client.py

```diff
@@ -160,10 +199,32 @@ def _send_batch(self, batch_data: Dict[str, Any]) -> Dict[str, Any]:
 
     def _get_batch(self, batch_id: str) -> Dict[str, Any]:
         response: Dict[str, Any] = self._request(
-            "GET", f"{self.endpoints.core}/api/v1/batches/{batch_id}"
+            "GET", f"{self.endpoints.core}/api/v2/batches/{batch_id}"
         )["data"]
         return response
 
+    def _get_batch_jobs(self, batch_id: str) -> list[Dict[str, Any]]:
+        return self._request_all_pages(
+            "GET",
+            f"{self.endpoints.core}/api/v2/jobs",
+            params={
+                "batch_id": batch_id,
+                "order_by": "ordered_id",
+                "order_by_direction": "ASC",
+            },
+        )
+
+    def _get_job_results(self, job_id: str) -> Any:
```

Comment:
Can we restrict the return type?

Reply:
In the current method, we are using `_request` to download results from S3. However, `_request` is using JSendPayload as a return typehint, which is not correct when we are not requesting core-api. That's why I used Any. 
But we can directly use `requests` in `_get_job_results` to have something like this:

 ```
   def _get_job_results(self, job_id: str) -> Optional[JobResult]:
        results_link = self._request(
            "GET", f"{self.endpoints.core}/api/v1/jobs/{job_id}/results_link"
        )["data"]["results_link"]
        if results_link:
            try:
                response = requests.get(results_link)
                response.raise_for_status()
                data = response.json()
                return JobResult(
                    raw=data.pop("raw", None), counter=data.pop("counter", None), **data
                )
            except HTTPError:
                pass
        return None
```
And we can, of course, have another internal method that performs HTTP calls, which will be used by `_request` and `_get_job_results`

what do you think?

Reply:
ok for me. I don't think we should use our `_request` method since it is meant to call our APIs - it automatically injects auth in the headers and so on which we don't want to interact with s3